pkg/storage/azure/resolver.go

```diff
@@ -0,0 +1,126 @@
+// FILE: pkg/storage/azure/resolver.go
+package azure
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"terralist/pkg/storage"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/sas"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+)
+
+type Resolver struct {
+	ContainerName string
+	AccountName   string
+	AccountKey    string
+	Client        *azblob.Client
+
+	// DefaultAzureCredentials *azidentity.DefaultAzureCredential
+
+	DefaultCredentials bool
+}
+
+func (r *Resolver) Store(in *storage.StoreInput) (string, error) {
+	// Create a new block blob URL using the container URL and the specified key
+	key := fmt.Sprintf("%s/%s", in.KeyPrefix, in.FileName)
+
+	ctx := context.Background()
+	buffer := []byte(in.Content)
+
+	_, err := r.Client.UploadBuffer(ctx,
+		r.ContainerName,
+		key,
+		buffer,
+		nil)
+	if err != nil {
+		return "", fmt.Errorf("could not upload archive: %v", err)
+	}
+	return key, nil
+
+}
+
+func (r *Resolver) GetSASURL(blobName string) (string, error) {
+
+	if r.DefaultCredentials {
+		serviceClient := r.Client.ServiceClient()
+
+		// Prepare SAS Token Input field
+		now := time.Now().UTC().Add(-10 * time.Second)
+		expiry := now.Add(48 * time.Hour)
+		info := service.KeyInfo{
+			Start:  to.Ptr(now.UTC().Format(sas.TimeFormat)),
+			Expiry: to.Ptr(expiry.UTC().Format(sas.TimeFormat)),
+		}
+
+		// Get Delegation Key
+		udc, err := serviceClient.GetUserDelegationCredential(context.TODO(), info, nil)
+		if err != nil {
+			return "", fmt.Errorf("could not get UserDelegationCredential: %v", err)
+		}
+
+		sasQueryParams, err := sas.BlobSignatureValues{
+			Protocol:      sas.ProtocolHTTPS,
+			StartTime:     time.Now().UTC().Add(time.Second * -10),
+			ExpiryTime:    time.Now().UTC().Add(15 * time.Minute),
+			Permissions:   to.Ptr(sas.BlobPermissions{Read: true, List: true}).String(),
+			ContainerName: r.ContainerName,
+			BlobName:      blobName,
+		}.SignWithUserDelegation(udc)
+		if err != nil {
+			log.Fatal(err.Error())
```

Comment:
Why is this logging the error instead of returning it?
```suggestion
			return "", err.Error()
```

Reply:
yep sorry, fixed